src/app/modules/submission/submission.service.ts

```diff
@@ -318,3 +327,48 @@ export const copyPendingSubmissionToSubmissions = (
       })
     })
 }
+
+/**
+ * Validates that the attachments in a submission do not violate form-level
+ * constraints e.g. form-wide attachment size limit.
+ * @param parsedResponses Unprocessed responses
+ * @returns okAsync(true) if validation passes
+ * @returns errAsync(InvalidFileExtensionError) if invalid file extensions are found
+ * @returns errAsync(AttachmentTooLargeError) if total attachment size exceeds 7MB
+ */
+export const validateAttachments = (
+  parsedResponses: ParsedClearFormFieldResponse[],
+): ResultAsync<true, InvalidFileExtensionError | AttachmentTooLargeError> => {
+  const logMeta = { action: 'validateAttachments' }
+  const attachments = mapAttachmentsFromResponses(parsedResponses)
+  if (areAttachmentsMoreThan7MB(attachments)) {
```

Comment:
since this is being shared across email and storage, need to have separate limit for storage

Reply:
Ah, good catch, thanks!

Reply:
Fixed here: [31847c4](https://github.com/opengovsg/FormSG/pull/6595/commits/31847c41c2edd8514e97779195f46e2a7a239996)